fairworkflows/fairstep.py

```diff
@@ -32,23 +32,31 @@ class FairVariable:
         uri: Optionally pass a uri that the variable is referred to, the variable name will be
             automatically extracted from it. This argument is usually only used when we extract a
             variable from rdf)
-        type: The type of the variable (i.e. int, str, float etc.)
+        type: The type of the variable (i.e. int, str, float etc.
     """
-    def __init__(self, name: str = None, type: str = None, uri: str = None):
+    def __init__(self, name: str = None, computational_type: str = None, types = None, uri: str = None):
         if uri and name is None:
             # Get the name from the uri (i.e. 'input1' from http://example.org#input1)
             _, name = urldefrag(uri)
         self.name = name
-        self.type = type
+        self.computational_type = computational_type
+
+        if types is None:
+            self.types = []
+        else:
+            if isinstance(types, str):
```

Comment:
As a user I would also expect passing a single rdflib.URIRef to work. Can we allow that?

Reply:
Yes, good spot. I've added that now.